presto-main/src/main/java/io/prestosql/metadata/Metadata.java

```diff
@@ -588,4 +591,62 @@ default ResolvedFunction getCoercion(Type fromType, Type toType)
      * The method is used by the engine to determine if a materialized view is current with respect to the tables it depends on.
      */
     MaterializedViewFreshness getMaterializedViewFreshness(Session session, QualifiedObjectName name);
+
+    /**
+     * Redirect to another table or view. Returns the original table name if the redirection doesn't happen.
+     */
+    QualifiedObjectName redirectTable(Session session, QualifiedObjectName tableName, WarningCollector warningCollector);
+
+    /**
+     * Redirect to another pair of source and target names. Returns original table names if the redirection doesn't happen.
+     */
+    RenameRedirection redirectTableRename(Session session, QualifiedObjectName sourceTableName, QualifiedObjectName targetTableName, WarningCollector warningCollector);
+
+    class RenameRedirection
+    {
+        private final QualifiedObjectName sourceTableName;
+        private final QualifiedObjectName targetTableName;
+
+        public RenameRedirection(QualifiedObjectName sourceTableName, QualifiedObjectName targetTableName)
+        {
+            this.sourceTableName = requireNonNull(sourceTableName, "sourceTableName is null");
+            this.targetTableName = requireNonNull(targetTableName, "targetTableName is null");
+        }
+
+        public QualifiedObjectName getSourceTableName()
+        {
+            return sourceTableName;
+        }
+
+        public QualifiedObjectName getTargetTableName()
+        {
+            return targetTableName;
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (obj == this) {
+                return true;
+            }
+            if ((obj == null) || (getClass() != obj.getClass())) {
+                return false;
+            }
+            RenameRedirection o = (RenameRedirection) obj;
+            return Objects.equals(sourceTableName, o.sourceTableName) &&
+                    Objects.equals(targetTableName, o.targetTableName);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(sourceTableName, targetTableName);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "(" + sourceTableName + ", " + targetTableName + ")";
```

Comment:
include "RenameRedirection" (or at least "redirect") in the toString.

Also consider replacing `,` with `=>`

Reply:
Included "Rename" as the pair reflects a rename relationship